lib/Parse/Lexer.cpp

```diff
@@ -1731,20 +1799,19 @@ void Lexer::lexStringLiteral() {
                              replacement);
       }
 
-      // Is this the end of a multiline string literal?
-      if (MultilineString) {
-        if (*CurPtr == '"' && *(CurPtr + 1) == '"' && *(CurPtr + 2) != '"') {
-          CurPtr += 2;
-          formToken(tok::string_literal, TokStart, MultilineString);
-          if (Diags)
-            validateMultilineIndents(NextToken, Diags);
-          return;
-        }
-        else
-          continue;
+      // Is this the end of multiline/delimited string literal?
+      if (StringRef(CurPtr, BufferEnd - CurPtr).startswith(ExtraTermination)) {
```

Comment:
What about `#" "##" "#`? is this two `#" "#`s or one literal containing `' "##" '`?

Reply:
Two literals I think though others have questioned this. “# is the termination unless we make a rule it can’t be followed by a #

Reply:
I’ve added the rule that a raw string can’t be terminated with a # immediately after it which people seem to expect for strings such as `#" "##" “#` I’ll ask the forum if this makes sense.

Reply:
Evolution didn’t seem to keen on the idea just when I’ve decided this was the best way to go.